proposals/selfAssemblingModelHierarchy/README.md

```diff
@@ -0,0 +1,342 @@
+def Xform "NewYorkCity" (kind = "assembly") {
+    def Scope "Water" (kind = "group") {
+        def Mesh "EastRiver" {}
+        def Material "EastRiverMaterial" { ... }
+    }
+}
+```
+
+The "EastRiver" is geometry just inlined as a mesh into the "NewYorkCity" assembly with a material. This "incomplete" hierarchy is not incorrect.  It just means that there are leaf prims that are not `component` descendants.
+
+It's worth noting that in the current implementation without propagation, `kind` will be checked on the `EastRiver` and `EastRiverMaterial` scopes. **If we can signal that propagation should end on these scopes, then we can provide a path for self assembling model hierarchies without additional `kind` reads.**
+
+#### Variant #1: Introduce a new `auto` kind which enables propagation
+`auto` becomes `kind`'s new fallback value. The "EastRiver" and other members of the incomplete hierarchy could be explicitly tagged with `kind=""` to preserve current behavior. Propagation for `group` or `assembly` kinds is `group`.
+
+#### Variant #2: Use authored state to control propagation
+Only propagate when `kind` is unauthored. This is effectively the same as `auto` in that "EastRiver" can be tagged with `kind=""` to suppress propagation.
+
+In user interfaces, it may be hard to disambiguate between unauthored (and propagating) and authored (not propagating) `kind` state. There would also not be way to author a value to re-enable propagation, but this may be fine.
+
+#### Variant #3: Use `subcomponent` (or some other kind) to prune traversal
+Despite its name, there are no rules that `subcomponent` prims are descendants of `component` prims. One could use that or some other kind to meaningfully stop traversal. `kind=none` (or `kind=terminal`) has been another suggestion. Even if variant #1 or #2 were adopted, this would still be a valid way to prune traversal, it just wouldn't be the only (or preferred) way.
+
+### Instancing
+Propagation when `instanceable=True` is complicated, as a prim may have multiple parents with different model hierarchy validity. There are actually [bugs](https://github.com/PixarAnimationStudios/OpenUSD/issues/2406) with this today.
+
+The OpenUSD development team proposes that prim flags are considered in concert with composition arcs when constructing prototypes to deal with this ambiguity.
+
+Until that fix is available, the best that can be done in both the current state of the world and with self propagating hierarchies is to encourage users to be intentional about model hierarchy when authoring `instanceable` to avoid violating the hierarchy's continuity.
+
+### Additional API
+To better clarify what it means to be a "model group", this proposal recommends deprecating `IsGroup` in favor of `MayContainComponentModel`.
+
+This proposal also recommends `IsModel` should be deprecated in favor of `MayContainOrIsComponentModel` and should be equivalent to (`MayContainComponentModel() || IsComponent()`).
```

Comment:
I'm not averse to introducing these API's, but I don't think the rest of the proposal makes the existing `IsGroup()`, `IsModel()`, or `IsComponent()` any less correct/relevant (and all already **effective**, computed results rather than simply authored kind), so deprecating the existing methods feels incorrect.

Noting also that I am conflicted, and curious of others' thoughts, on whether, should we pursue this proposal, there would want to be updates to the Model Hierarchy description in the Kind developer documentation?  It talks about model hierarchy, but falls short of fully prescribing the population rules (which are out of scope for this low-level module anyways).

For sure we would want to add a User Guide for model hierarchy as part of this work!